experiment/autobumper/bumper/bumper.go

```diff
@@ -518,14 +518,59 @@ func GitCommitAndPush(remote, remoteBranch, name, email, message string, stdout,
 }
 
 // GitPush push the changes to the given remote and branch.
-func GitPush(remote, remoteBranch string, stdout, stderr io.Writer) error {
+func GitPush(remote, remoteBranch string, stdout, stderr io.Writer, workingDir string) error {
 	logrus.Info("Pushing to remote...")
-	if err := Call(stdout, stderr, "git", "push", "-f", remote, fmt.Sprintf("HEAD:%s", remoteBranch)); err != nil {
+	gitArgs := []string{}
+	if workingDir != "" {
+		gitArgs = append(gitArgs, fmt.Sprintf("--git-dir=%s/.git", workingDir), fmt.Sprintf("--work-tree=%s", workingDir))
+	}
+	gitPushArgs := append(gitArgs, "push", "-f", remote, fmt.Sprintf("HEAD:%s", remoteBranch))
+	if err := Call(stdout, stderr, "git", gitPushArgs...); err != nil {
 		return fmt.Errorf("failed to git push: %w", err)
 	}
 	return nil
 }
 
+// RunAndCommitIfNeeded makes a commit in the workingDir if there are
+// any changes resulting from the command execution. Returns true if a commit is made
+func RunAndCommitIfNeeded(stdout, stderr io.Writer, author, cmd string, args []string, workingDir string) (bool, error) {
+	fullCommand := fmt.Sprintf("%s %s", filepath.Base(cmd), strings.Join(args, " "))
+
+	logrus.Infof("Running: %s", fullCommand)
+	if err := Call(stdout, stderr, cmd, args...); err != nil {
+		return false, fmt.Errorf("failed to run %s: %w", fullCommand, err)
+	}
+
+	changed, err := HasChanges()
+	if err != nil {
+		return false, fmt.Errorf("error occurred when checking changes: %w", err)
+	}
+
+	if !changed {
+		logrus.WithField("command", fullCommand).Info("No changes to commit")
+		return false, nil
+	}
+
+	gitCmd := "git"
+	gitArgs := []string{}
+	if workingDir != "" {
```

Comment:
If this might be used across even more functions in the future is it worth making something like (below)?
```
type GitCommand struct {
         baseCommand       string
         workingDir               string
         args                          []string{}
}

func (gc GitCommand) Call(stdout, stderr io.Writer) error {
        	if gc.workingDir != "" {
		gc.args = append(gc.args, fmt.Sprintf("--git-dir=%s/.git", gc.workingDir), fmt.Sprintf("--work-tree=%s", gc.workingDir))
	}
	c := exec.Command(gc.baseCommand, gc.args...)
	c.Stdout = stdout
	c.Stderr = stderr
	return c.Run()
}
```